src/runtime/complex.cpp

```diff
@@ -228,6 +228,29 @@ static void _addFunc(const char* name, ConcreteCompilerType* rtn_type, void* com
     complex_cls->giveAttr(name, new BoxedFunction(cl));
 }
 
+Box* complexHash(BoxedComplex* self) {
+    if (!isSubclass(self->cls, complex_cls))
+        raiseExcHelper(TypeError, "descriptor '__hash__' requires a 'complex' object but received a '%s'",
+                       getTypeName(self));
+    long hashreal, hashimag, combined;
+    hashreal = _Py_HashDouble(self->real);
+    if (hashreal == -1)
+        return boxInt(-1);
```

Comment:
these should be `throwCapiException()` -- returning -1 is the capi way for hash functions to specify that they threw an exception.